src/controller/couponController.ts

```diff
@@ -0,0 +1,143 @@
+import { Request, Response } from 'express';
+import Coupon from '../database/models/couponEntity';
+import dbConnection from '../database';
+import errorHandler from '../middlewares/errorHandler';
+import { check, validationResult } from 'express-validator';
+import Product from '../database/models/productEntity';
+import crypto from 'crypto';
+import UserModel from '../database/models/userModel';
+
+const couponRepository = dbConnection.getRepository(Coupon);
+const productRepository = dbConnection.getRepository(Product);
+
+interface couponRequestBody {
+    description: string;
+    percentage: number;
+    expirationDate: Date;
+    applicableProducts: number[];
+}
+
+const createCouponRules = [
+    check('description').isLength({ min: 1 }).withMessage('coupon description is required'),
+    check('percentage').isInt({ min: 0, max: 100 }).withMessage('percentage must be between 0 and 100'),
+    check('expirationDate').isDate().withMessage('expiration date must be a valid date'),
+    check('applicableProducts').isArray().withMessage('applicable products must be an array of product ids'),
+]
+class CouponController {
+    // GET /coupons
+    public getAllCoupons = errorHandler(async (req: Request, res: Response) => {
```

Comment:
This function seems to get all coupons. Shouldn't you maybe have another function that retrieves the coupons that belong to a specific seller? and not all coupons?

Reply:
I'm currently solving that and in 26 minutes, I will be done

Reply:
That has been done